garmin_txt.cc

```diff
@@ -1230,23 +1216,19 @@ parse_track_header()
 static void
 parse_route_waypoint()
 {
-  char* str;
-  int column = -1;
-  Waypoint* wpt = nullptr;
-
-  bind_fields(rtept_header);
+  // An example of how most of these parse_foo()'s could look if brought to c++.
+  auto tokens = current_line_text.split("\t");
+  if (tokens.size() < 1) {
+    Fatal() << MYNAME ": Route waypoint at " << current_line;
+  }
 
-  while ((str = csv_lineparse(nullptr, "\t", "", column++))) {
-    int field_no = header_fields[rtept_header][column];
-    switch (field_no) {
-    case 1:
-      is_fatal((*str == '\0'), MYNAME ": Route waypoint without name at line %d!\n", current_line);
-      wpt = find_waypt_by_name(str);
-      is_fatal((wpt == nullptr), MYNAME ": Route waypoint \"%s\" not in waypoint list (line %d)!\n", str, current_line);
-      wpt = new Waypoint(*wpt);
-      break;
-    }
+  auto wpt = find_waypt_by_name(tokens.at(1));
+  if (wpt == nullptr) {
+    Fatal() << MYNAME ": Route waypoint " << tokens.at(1) << 
+      " not in waypoint list (line " << current_line << ")!\n";
   }
+
+  wpt = new Waypoint(*wpt);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Either the condition 'wpt==nullptr' is redundant or there is possible null pointer dereference: wpt.](https://app.codacy.com/app/gpsbabel/gpsbabel/pullRequest?prid=4695230)